pvlib/horizon.py

```diff
@@ -0,0 +1,459 @@
+        else:
+            F_M += d_ne
+            d_ne += 2
+            y += 1
+        d_e += 2
+        d_ne += 2
+        x += 1
+        points.extend(_point_symmetry(x, y))
+    points = np.array(points)
+    newx = points[:, 0] + center[0]
+    newy = points[:, 1] + center[1]
+    return np.vstack((newx, newy))
+
+
+def _pol2cart(r, theta):
+    r'''Converts polar to cartesian
+    Parameters
+    ----------
+    r : np.array
+        r values for the points
+    theta: np.array
+        theta values for the points
+
+    Returns
+    -------
+    x : np.array
+        x values for the points
+    y : np.array
+        y values for the points
+
+    '''
+
+    z = r * np.exp(1j * theta)
+    x, y = z.real, z.imag
+    return x, y
+
+
+def _cart2pol(x, y):
+    r'''Converts cartesian to polar
+    Parameters
+    ----------
+    x : np.array
+        x values for the points
+    y : np.array
+        y values for the points
+
+    Returns
+    -------
+    r : np.array
+        r values for the points
+    theta: np.array
+        theta values for the points
+
+    '''
+
+    z = x + y * 1j
+    r, theta = np.abs(z), np.angle(z)
+    return r, theta
+
+
+def _sort_circ(pts, center=(0, 0), az_len=360):
+    r'''Sort and resample points on a circle such that the zeroth element is
+    due east and the points move around the circle counter clockwise.
+    While in polar domain, resample points using FFT to
+    obtain desired number of bins,typically 360, for
+    degrees around the circle.
+
+    Parameters
+    ----------
+    pts : np.array
+        Array of shape (n, 2) of points in Cartesian system
+    center : tuple
+        Center (x,y) of the circle rasterized by pts
+    az_len : numeric
+        Desired number of points in the rasterized circle.
+
+    Returns
+    -------
+    points : np.array
+        Sorted and resampled points that comprise the circle
+    '''
+    pts[0] -= center[0]
+    pts[1] -= center[1]
+    r, theta = _cart2pol(pts[0], pts[1])
+    stacked = np.vstack((theta, r))
+    sort = np.sort(stacked, axis=1)
+    sort = resample(sort, az_len, axis=1)
+    theta = sort[0]
+    r = sort[1]
+    x, y = _pol2cart(r, theta)
+    x += center[0]
+    y += center[1]
+    return np.vstack((x, y)).astype(int)
+
+
+def horizon_map(dem_pixel, elevation, dem_res=30.0,
+                view_distance=500, az_len=36):
+    r"""Finds the horizon at point on a dem in pixel coordinates dem_pixel
+
+    Parameters
+    ----------
+    dem_pixel : tuple (int,int)
+        Point on the DEM expressed in pixel coordinates
+    elevation : np.array
+        nxn DEM of elevation values
+    dem_res : float
+        Resolution of the DEM. The default is SRTM 30m
+    view_distance : int
+        Radius of the area of consideration.
+    az_len : int
+        Number of samples on the perimeter of the circle.
+
+    Returns
+    -------
+    azimuth : np.array
+        Numpy array of shape (az_len). Linearly spaced from [0,360]
```

Comment:
Can `azimuth` include the endpoint 360?  That would be a duplicate. Maybe '[0, 360)'